tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntryPathComparer.cs

```diff
@@ -0,0 +1,260 @@
+        /// </summary>
+        private int CompareByServiceLabel(CodeownersEntry x, CodeownersEntry y)
+        {
+            // Get the primary service context for each entry (service label first, then PR label)
+            string primaryLabelX = GetPrimaryServiceContext(x);
+            string primaryLabelY = GetPrimaryServiceContext(y);
+
+            // If both have primary labels, compare them alphabetically
+            if (!string.IsNullOrEmpty(primaryLabelX) && !string.IsNullOrEmpty(primaryLabelY))
+            {
+                return string.Compare(primaryLabelX, primaryLabelY, StringComparison.OrdinalIgnoreCase);
+            }
+
+            // If only one has a primary label, prioritize the one with the label
+            if (!string.IsNullOrEmpty(primaryLabelX) && string.IsNullOrEmpty(primaryLabelY))
+            {
+                return -1; // x has primary label, y doesn't - x comes first
+            }
+            if (string.IsNullOrEmpty(primaryLabelX) && !string.IsNullOrEmpty(primaryLabelY))
+            {
+                return 1; // y has primary label, x doesn't - y comes first
+            }
+
+            // If neither has labels, they're equal at this level
+            return 0;
+        }
+
+        /// <summary>
+        /// Get the primary service context for an entry. This prioritizes ServiceLabel but falls back to the first PRLabel.
+        /// This ensures entries with combined PR labels like "AI Model Inference %AI Projects" are grouped with 
+        /// individual "AI Model Inference" entries. For hierarchical services like "Communication - Call Automation",
+        /// this extracts the base service name "Communication" for grouping.
+        /// </summary>
+        private string GetPrimaryServiceContext(CodeownersEntry entry)
+        {
+            // First priority: Service labels
+            if (entry.ServiceLabels?.Count > 0)
+            {
+                var serviceLabel = entry.ServiceLabels[0].TrimStart('%').Trim();
+                if (!string.IsNullOrEmpty(serviceLabel))
+                {
+                    return serviceLabel;
+                }
+            }
+
+            // Second priority: PR labels (use the first one as primary context)
+            if (entry.PRLabels?.Count > 0)
+            {
+                var prLabel = entry.PRLabels[0].TrimStart('%').Trim();
+                if (!string.IsNullOrEmpty(prLabel))
+                {
+                    return prLabel;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Determines if a path is a catch-all pattern that should appear at the beginning of a section.
+        /// Catch-all patterns are simple directory paths like "/sdk/" that don't contain wildcards
+        /// but serve as fallback patterns for entire directory trees.
+        /// </summary>
+        private bool IsCatchAllPattern(string path)
```

Comment:
```suggestion 
        private static bool IsCatchAllPattern(string path)
```

This method doesn't depend use any variables from a `CodeownersEntryPathComparer` instance. It can be static.